src/peripheral/nvic.rs

```diff
@@ -65,9 +65,36 @@ pub struct RegisterBlock {
     /// so convenient byte-sized representation wouldn't work on that
     /// architecture.
     pub ipr: [RW<u32>; 8],
+
+    #[cfg(not(armv6m))]
+    reserved5: [u32; 208],
+
+    #[cfg(armv6m)]
+    reserved5: [u32; 696],
+
+    #[cfg(not(armv6m))]
+    /// Software Trigger Interrupt
+    pub stir: WO<u32>,
 }
 
 impl NVIC {
+    #[cfg(not(armv6m))]
+    /// Request an IRQ in software
+    ///
+    /// Writing a value to the INTID field is the same as manually pending an interrupt by setting
+    /// the corresponding interrupt bit in an Interrupt Set Pending Register. This is similar to
+    /// `set_pending`.
```

Comment:
Could you include in the documentation why would one prefer this over `set_pending` and also mention that this API is not available on ARMv6-M.

Reply:
I'll be honest... I don't know why to choose one vs the other. The reference manuals don't make the different clear. :) I'll add a note about not being available on armv6m